### PR TITLE
feat: Include repo as a tag

### PR DESCRIPTION
**Pull request by armenzg**
This is to support ingesting jobs from different repos, thus, enabling different teams to build their own dashboards.

There are other changeds like:

- Rename head_sha tag to commit
- pull_request tag to be just a number
- Include pull_request as part of the extra metadata to enable navigating to PR on Github

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,4 +76,4 @@ Table of commands:
 
 ## Sentry staff info
 
-Google Cloud Build will automatically build a Docker image when the code merges on `master`. Logging to Google Cloud Run and deploy the latest image.
+Google Cloud Build will automatically build a Docker image when the code merges on `main`. Logging to Google Cloud Run and deploy the latest image.
```
Comment (armenzg) [inline README.md:79]: Typo!

File: src/cli.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import sys
 
-from trace import get, send_trace
+from github_sdk import get, send_trace
```
Comment (armenzg) [inline src/cli.py:3]: I forgot to handle this in previous PRs.

File: src/github_sdk.py [hunk 3]
```diff
@@ -71,19 +71,29 @@ def send_envelope(trace):
 def get_extra_metadata(job):
```
Comment (armenzg) [inline src/github_sdk.py:71]: All of the changes in this file can more easily be view if you look at the `trace.json` changes.

The changes in here are to make the function `_generate_trace` a lot simpler. For instance, now, I can just do this:

```python
    transaction["tags"] = meta["tags"]
    transaction["contexts"]["trace"]["data"] = meta["data"]
```

File: src/github_sdk.py [hunk 4]
```diff
@@ -71,19 +71,29 @@ def send_envelope(trace):
...
     runs = get(job["run_url"]).json()
     workflow = get(runs["workflow_url"]).json()
+    repo = runs["head_repository"]["full_name"]
```
Comment (armenzg) [inline src/github_sdk.py:74]: This is part of the actual feature on this repo.

File: src/github_sdk.py [hunk 5]
```diff
@@ -71,19 +71,29 @@ def send_envelope(trace):
...
     meta = {
-        "workflow_name": workflow["name"],
-        # .github/workflows/foo.yml -> foo.yml
-        "workflow_path": workflow["path"].rsplit("/")[-1],
-        "head_branch": runs.get("head_branch"),
-        "head_sha": runs.get("head_sha"),
+        # "workflow_name": workflow["name"],
```
Comment (armenzg) [inline src/github_sdk.py:76]:
```suggestion
```

File: src/github_sdk.py [hunk 6]
```diff
@@ -71,19 +71,29 @@ def send_envelope(trace):
 def get_extra_metadata(job):
     runs = get(job["run_url"]).json()
     workflow = get(runs["workflow_url"]).json()
+    repo = runs["head_repository"]["full_name"]
     meta = {
-        "workflow_name": workflow["name"],
```
Comment (armenzg) [inline src/github_sdk.py:75]: This is not used anywhere.

File: src/github_sdk.py [hunk 7]
```diff
@@ -71,19 +71,29 @@ def send_envelope(trace):
...
-        # .github/workflows/foo.yml -> foo.yml
-        "workflow_path": workflow["path"].rsplit("/")[-1],
```
Comment (armenzg) [inline src/github_sdk.py:77]: These 3 values are now within the `tag` key below.

File: src/github_sdk.py [hunk 8]
```diff
@@ -71,19 +71,29 @@ def send_envelope(trace):
...
-        "head_branch": runs.get("head_branch"),
-        "head_sha": runs.get("head_sha"),
+        # "workflow_name": workflow["name"],
         "author": runs["head_commit"]["author"],
-        "run_attempt": runs["run_attempt"],
+        # https://getsentry.atlassian.net/browse/TET-22
+        # Tags are not linkified externally, plain text data can be selected in browsers and opened
+        "data": {
+            "job": job["html_url"],
+        },
```
Comment (armenzg) [inline src/github_sdk.py:82]: I've moved the `data` key from the other function in here.

File: src/github_sdk.py [hunk 9]
```diff
@@ -71,19 +71,29 @@ def send_envelope(trace):
...
+    repo = runs["head_repository"]["full_name"]
     meta = {
-        "workflow_name": workflow["name"],
-        # .github/workflows/foo.yml -> foo.yml
-        "workflow_path": workflow["path"].rsplit("/")[-1],
-        "head_branch": runs.get("head_branch"),
-        "head_sha": runs.get("head_sha"),
+        # "workflow_name": workflow["name"],
         "author": runs["head_commit"]["author"],
-        "run_attempt": runs["run_attempt"],
```
Comment (armenzg) [inline src/github_sdk.py:81]: Moved down below within `tag` key.

File: src/github_sdk.py [hunk 10]
```diff
@@ -71,19 +71,29 @@ def send_envelope(trace):
...
+            # It allows querying jobs within the same workflow (e.g. foo.yml)
+            "workflow": workflow["path"].rsplit("/")[-1],
+        },
     }
     if runs.get("pull_requests"):
-        meta[
-            "pull_request"
-        ] = f'https://github.com/{runs["head_repository"]["full_name"]}/pull/{runs["pull_requests"][0]["number"]}'
+        pr_number = runs["pull_requests"][0]["number"]
+        meta["data"]["pr"] = f"https://github.com/{repo}/pull/{pr_number}"
```
Comment (armenzg) [inline src/github_sdk.py:95]: This is new, it allows for selecting text in an issue and then being able to navigate to the PR.

For [instance](https://sentry.io/organizations/sentry-test/performance/armen-test-project:dfb378a450374d95853e546f94e88a8b/?project=6374757&query=transaction.duration%3A%3C15m&statsPeriod=24h&transaction=create+frontend+bundle&unselectedSeries=p100%28%29):
image

File: src/github_sdk.py [hunk 11]
```diff
@@ -71,19 +71,29 @@ def send_envelope(trace):
...
+        meta["tags"]["pull_request"] = pr_number
```
Comment (armenzg) [inline src/github_sdk.py:96]: Before, the tag was a URL, it is now a number:
image

File: src/github_sdk.py [hunk 12]
```diff
@@ -137,26 +147,11 @@ def _generate_spans(steps, parent_span_id, trace_id):
...
-        "run_attempt": meta["run_attempt"],
-        # To filter jobs that run on a specific sha
-        "head_sha": meta["head_sha"],
-        # Workflow name as a tag. It allows querying jobs within the same workflow
-        "workflow": meta["workflow_path"],
-    }
-    if meta.get("pull_request"):
-        transaction["tags"]["pull_request"] = meta["pull_request"]
+    transaction["tags"] = meta["tags"]
+    transaction["contexts"]["trace"]["data"] = meta["data"]
```
Comment (armenzg) [inline src/github_sdk.py:152]: This function is now less convoluted.

File: tests/fixtures/jobA/trace.json [hunk 13]
```diff
@@ -54,9 +55,10 @@
   "start_timestamp": "2022-04-06T19:52:17Z",
   "tags": {
     "branch": "ahmed/add-naming-layer",
-    "head_sha": "fd976218a94d8a0cd203c711ea7dbe68c573ef4d",
+    "commit": "fd976218a94d8a0cd203c711ea7dbe68c573ef4d",
     "job_status": "success",
-    "pull_request": "https://github.com/getsentry/sentry/pull/33347",
+    "pull_request": 33347,
+    "repo": "getsentry/sentry",
```
Comment (armenzg) [inline tests/fixtures/jobA/trace.json:61]: This is the actual feature. Everything else is small polishes.

**Review by armenzg — COMMENTED** [review]
See `trace.json` first when reviewing.

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**